packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -165,7 +170,9 @@ export class DefaultCrudRepository<T extends Entity, ID>
     ForeignKeyType
   >(
     relationName: string,
-    targetRepo: EntityCrudRepository<Target, TargetID>,
+    targetRepo:
```

Comment:
Due to the issue of dependency cycle, a getter needs to be used and eventually invoked to resolve the target repository. I'm thinking that we need a separate decorator as per @raymondfeng's suggestion. Something like `@getRepository`?

Reply:
I wonder if `@repository` can be extended to provide us that support?

Reply:
I think it's possible. Let me tinker around with it

Reply:
Visited the code for `@repository`. Here are the options that I see:

1. pass in an option along the lines of `resolveCircularDependency`. When enabled, `@repository` can call `inject.getter` underneath instead of `inject`
2. Detect if the prototype passed in to the decorator inherits from `DefaultCrudRepository` and then call `inject.getter`. This allows the UX to remain the same as before, but there might be a headache maintaining this with other future implementations of repositories. I also don't know easy it'd be to figure out whether the prototype belongs to `DefaultCrudRepository` or not

Reply:
I see. @raymondfeng @bajtos thoughts?